### PR TITLE
feat(NET-554): add option to upgrade client version in hosts table

### PR DESCRIPTION
**Pull request by pwilson77**

### CODE REVIEW

File: src/pages/hosts/HostsPage.tsx [hunk 1]
```diff
@@ -207,6 +207,37 @@ export default function HostsPage(props: PageProps) {
...
+  const confirmUpgradeClient = useCallback(
+    async (host: Host) => {
+      Modal.confirm({
+        title: 'Upgrade host version',
+        content: (
+          <>
+            <Row>
+              <Col xs={24}>
+                <Typography.Text>
+                  Are you sure you want to upgrade the version of this host {host.name}?
```
Comment (Aceix) [inline src/pages/hosts/HostsPage.tsx:219]: is it possible to know the version user is upgrading to? would be helpful to show

File: src/pages/hosts/HostsPage.tsx [hunk 2]
```diff
@@ -207,6 +207,37 @@ export default function HostsPage(props: PageProps) {
...
+          } catch (err) {
+            notify.error({
+              message: 'Failed to upgrade client version',
+              description: extractErrorMsg(err as any),
+            });
+          }
+        },
+      });
+    },
+    [notify, store.nodes],
```
Comment (Aceix) [inline src/pages/hosts/HostsPage.tsx:238]: remove `store.nodes`

File: src/pages/hosts/HostsPage.tsx [hunk 3]
```diff
@@ -353,6 +384,15 @@ export default function HostsPage(props: PageProps) {
...
                   },
+                  {
+                    key: 'upgrade',
+                    label: 'Upgrade Version',
+                    disabled: host.version === store.serverConfig?.Version,
+                    onClick: (ev) => {
+                      ev.domEvent.stopPropagation();
+                      confirmUpgradeClient(host);
+                    },
+                  },
```
Comment (Aceix) [inline src/pages/hosts/HostsPage.tsx:395]: on L414, add missing dependencies to the array 'confirmUpgradeClient', 'requestHostPull', and 'store.serverConfig?.Version'

**Review by Aceix — CHANGES_REQUESTED** [review]

File: src/pages/hosts/HostsPage.tsx (hunk 1 above)
Comment (Aceix) [reply src/pages/hosts/HostsPage.tsx:219]: judging from the backend PR, it up/downgrades to the server version. so you can specify this in the ui as well

Comment (pwilson77) [reply src/pages/hosts/HostsPage.tsx:219]: if its upgrading / downgrading I guess I need to know the next version

Comment (Aceix) [reply src/pages/hosts/HostsPage.tsx]: seen. actually, it only upgrades upto server version. let's specify the version then

**Review by Aceix — CHANGES_REQUESTED** [review]
pending 2 changes

File: src/pages/hosts/HostsPage.tsx (hunk 2 above)
Comment (pwilson77) [reply src/pages/hosts/HostsPage.tsx:238]: done

File: src/pages/hosts/HostsPage.tsx (hunk 3 above)
Comment (pwilson77) [reply src/pages/hosts/HostsPage.tsx:395]: done

**Review by Aceix — APPROVED** [review]

**Pull request merged**